blog/content/post/2022-03-15-fobnail_3rd_phase.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Fobnail Token - platform provisioning
+abstract: 'The Fobnail Token is an open-source hardware USB device that helps to
+           determine the integrity of the system. The purpose of this blog post
+           is to present the development progress of this project. This phase
+           was focused on platform provisioning.'
+cover: /covers/usb_token.png
+author: krystian.hebel
+layout: post
+published: true
+date: 2022-03-15
+archives: "2022"
+
+tags:
+  - fobnail
+  - firmware
+  - tpm
+  - attestation
+categories:
+  - Firmware
+  - Security
+
+---
+
+# About the Fobnail Token project
+
+The Fobnail Token is a project that aims to provide a reference architecture for
+building offline integrity measurement servers on the USB device and clients
+running in Dynamically Launched Measured Environments (DLME). It allows the
+Fobnail owner to verify the trustworthiness of the running system before
+performing any sensitive operation. This project was founded by [NlNet
+Foundation](https://nlnet.nl/). More information about the project can be found
+in the [Fobnail documentation](https://fobnail.3mdeb.com/). Also, make sure to
+read other posts related to this project by visiting
+[fobnail](https://www.blog.3mdeb.com/tags/fobnail/) tag.
+
+# Keys used in communication between Fobnail Token and attester
+
+In current phase two key pairs are used during communication, those are AIK and
+EK. Both belong to TPM installed on attester and their private keys never have
```

Comment:
But this is not a limitation, there are cases when key has to be exported. One of such cases is moving key to another TPM, or it can be used to sign data in code instead of passing it to TPM, which is usually much faster.

Reply:
(Proper) AIK can't be loaded on another TPM because it is created under given EK, which in turn can't be exported. You can manually change flags and load "the same" key (same `e` and `n`), but it would give different hash and different `loaded_key_name` in turn. This is one of reasons why Make/ActivateCredential is enough to prove that those two keys (or objects in general) are connected.

I will try to add a note that mentions that private key is temporarily saved in RAM during creation, but I don't want to spend too much content on that matter.

Reply:
(Proper) AIK can't be loaded on another TPM because it is created under given EK, which in turn can't be exported. You can manually change flags and load "the same" key (same `e` and `n`), but it would give different hash and different `loaded_key_name` in turn. This is one of reasons why Make/ActivateCredential is enough to prove that those two keys (or objects in general) are connected.

I will try to add a note that mentions that private key is temporarily saved in RAM during creation, but I don't want to spend too much content on that matter.